src/gsSolver/gsParameterDomainPreconditioners.h

```diff
@@ -0,0 +1,108 @@
+/** @file gsParameterDomainPreconditioners.h
+
+    @brief Provides preconditioners that live on the parameter domain.
+
+    This file is part of the G+Smo library.
+
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+    Author(s): S. Takacs, C. Hofreither
+*/
+#pragma once
+
+#include <gsAssembler/gsAssembler.h>
+#include <gsSolver/gsLinearOperator.h>
+
+
+namespace gismo
+{
+
+/// @brief Provides preconditioners that live on the parameter domain.
+///
+/// This class provides efficient preconditioners based on the parameter domain,
+/// assuming to have \a gsTensorBasis.
```

Comment:
What does it mean that they "live on the parameter domain"? 

The assumption of tensorbasis should be enforced in the code

Reply:
"live on the parameter domain" means that we ignore the contributions from the geometry transformation function.

Reply:
the tensor-bases do not have a common class, so it practically impossible to do so.

it will be even more impossible, if (what we are doing in devel) we want to apply such parameter-domain based preconditioners in a multi-patch manner because gsMultiBasis only gives us gsBasis. This means that the caller has the burden of casting around.